src/design-pages/passkey_authentication.rst

```diff
@@ -0,0 +1,318 @@
+
+Problem statement
+-----------------
+The use of new tools to authenticate users, such as 2FA, U2F and FIDO2, is
+becoming increasingly popular and currently SSSD doesn’t provide any way to use
+the latter. This diminishes the value provided by SSSD in some environments
+like big companies and governments, where the usage of these authentication
+mechanisms is becoming a common pattern.
+
+SSSD should provide a way to authenticate users that use a passkey. This would
+add more value by attracting the types of users mentioned in the previous
+paragraph.
+
+For the purpose of this feature, passkey is a FIDO2 compatible device supported
+by the libfido2 library.
+
+Use cases
+---------
+As a user, I want to walk to any system on a network and log in (desktop and/or
+terminal) using a passkey device connected to the system. The user credentials
+will be stored in a central management system like FreeIPA or Active Directory.
+
+Solution overview
+------------------------
+The objective is to use a passkey to locally authenticate a user against 
+centralized identity management system. For that purpose an integration with an
+LDAP server like AD or FreeIPA is needed.
+
+First, the key is registered in the LDAP server. Then, the SSSD clients should
+get the user credentials from this server, including the user key mapping data.
+On top of that, SSSD should communicate with the key to get the information
+stored in the device. Once this is done, SSSD should use all this data to
+authenticate the user.
+
+The biggest advantage of this approach is the usage of a central repository to
+store the user's credentials. In the long term this gives more flexibility as
+the users and keys can be authorized and revoked by the system administrator
+easily.
+
+Existing solution comparison
+****************************
+Existing Yubikey devices have already been enabled in RHEL and Fedora but the
+approach is completely different. The existing solution uses
+`OATH-TOTP <https://en.wikipedia.org/wiki/Time-based_one-time_password>`__ and
+`OATH-HOTP <https://en.wikipedia.org/wiki/HMAC-based_one-time_password>`__ to
+generate a one-time password (OTP) that is sent to the server as a second
+factor authentication. In other words, the authentication checks happen on the
+server. Moreover, this solution relies on a plugin for the 389 directory
+server, so it only works with IPA for IPA users.
+
+The approach explained in this document relies on the 
+`FIDO2 <https://en.wikipedia.org/wiki/FIDO2_Project>`__ standard, which uses a
+public-private key scheme to perform the authentication checks on the client.
+Besides, the solution is general enough to accept different LDAP servers like
+AD and IPA.
+
+Another difference is that with the current IPA feature, the first and second
+factor can also be used for doing a direct LDAP bind. This will not be possible
+with the new passkey scheme. Direct LDAP bind support will not be in the scope
+of this implementation.
```

Comment:
I found this paragraph confusing. Is direct bind with FIDO supported on IPA but unsupported on plain LDAP? Or do you refer to OTP?

Reply:
I think this paragraph can be dropped, it does not add anything to clarify the implementation.

Reply:
Direct bind isn't supported in the scope of this feature. In this section of the document I'm explaining the differences between the already implement OTP feature and this one.

It's kind of confusing because the FIDO2 keys provided by several vendors also contain other protocols (OTP, PIV, etc.). So, even if we refer to them as FIDO2 keys, they provide FIDO2 and other mechanisms. When I started writing this document and asked for feedback, some people didn't understand why the actual implementation (OTP) wasn't good enough. Our users were already using FIDO2 keys to authenticate, why did we need another implementation? That's why I wrote this section.

Reply:
I think what is needed here is an explicit statement that only FIDO2 part is supported here. If a hardware token implements other authentication mechanisms aside from FIDO2, these aren't considered by the passkey authentication feature. What is done by the FreeIPA server side should not be documented here.

Reply:
I dropped this section from the document as it doesn't make sense in this context and I added an explanation in the `Problem statement` to indicate that his feature is FIDO2 only.